src/helm/benchmark/scenarios/cleva_scenario.py

```diff
@@ -0,0 +1,1530 @@
+                        [
+                            template["item_template"].format(
+                                **i, idx=self.index_mapping(idx, template["item_index"])  # type: ignore
+                            )
+                            for idx, i in enumerate(data)
+                        ]
+                    )
+                # If each entry is a `str`, apply the template independently
+                else:
+                    # In this case, we reserve a default `item` key to hold each entry
+                    return template["item_separator"].join(
+                        [
+                            template["item_template"].format(
+                                item=i, idx=self.index_mapping(idx, template["item_index"])
+                            )
+                            for idx, i in enumerate(data)
+                        ]
+                    )
+            else:
+                raise ValueError(f"Unsupported input: {data}")
+        # Simple copying if template is None
+        elif template is None:
+            assert isinstance(data, str), "If there is no template, the data should be a string."
+            return data
+        else:
+            raise NotImplementedError(f"Unsupported template {template}")
+
+    def _mapping_all(self, data: Dict[str, Any], mapping_dict: Dict[str, Dict[str, str]]) -> None:
+        """
+        This function subsitute values in `data` according to the mapping defined in `mapping_dict` with the same
+        key/field.
+
+        Each field in `data` must have one of the following types: `str`, `Dict[str, str]`, `List[str]`, and
+        `List[Dict[str, str]]`.
+
+        Note that this is an in-place operation.
+        """
+        for k, d in mapping_dict.items():
+            for _name in data:
+                # If the value is a string, we directly map the result
+                if isinstance(data[_name], str):
+                    if _name == k:
+                        # Only perform the substitution if the keys in the `sample` match `mapping_dict`
+                        data[_name] = d[data[_name]]
+                # If the value is a dict, we map the value of its key-value pairs
+                elif isinstance(data[_name], dict):
+                    for _k in data[_name]:
+                        # Only perform the subsitution if the keys match
+                        if _k == k:
+                            assert isinstance(
+                                data[_name][_k], str
+                            ), "We only support mapping data with type `Dict[str, str]`"
+                            data[_name][_k] = d[data[_name][_k]]
+                # If the value is a list, then look into its entries
+                elif isinstance(data[_name], list):
+                    assert len(data[_name]) > 0, f"The length of {_name} must be larger than 0."
+                    if isinstance(data[_name][0], int):
```

Comment:
This only maps the first element of the index; do you need to also map the remaining elements?

I'm also wondering if you can use recursion handle the inner elements.

Reply:
Actually, we use the first element to check the type to determine how the mapping should apply. The mapping will be applied to every element in the list, as shown in the following couple of lines.